src/ntlm_auth_wrap.c

```diff
@@ -345,37 +370,89 @@ char **argv, **envp;
+        char *s_nt_response;
+        char *s_ntlm_password;
+        char *i_request_nt_key;
+
+        cJSON *json = cJSON_CreateObject();
+        if (json == NULL) {
+            fprintf(stderr, "Error: could not create JSON object. Exiting.");
             exit(1);
         }
+        for (int i=1; i<argc; i++) {
+            if (strncmp(argv[i], "--username=", strlen("--username=")) == 0) {
+                cJSON_AddStringToObject(json, "username", argv[i]+strlen("--username="));
+            }
+            if (strncmp(argv[i], "--password=", strlen("--password=")) == 0) {
+                cJSON_AddStringToObject(json, "password", argv[i]+strlen("--password="));
+            }
+            if (strncmp(argv[i], "--request-nt-key", strlen("--request-nt-key")) == 0) {
+                cJSON_AddItemToObject(json, "request-nt-key", cJSON_CreateTrue());
+            }
+            if (strncmp(argv[i], "--challenge=", strlen("--challenge=")) == 0) {
+                cJSON_AddStringToObject(json, "challenge", argv[i]+strlen("--challenge="));
+            }
+            if (strncmp(argv[i], "--nt-response=", strlen("--nt-response=")) == 0) {
+                cJSON_AddStringToObject(json, "nt-response", argv[i]+strlen("--nt-response="));
+            }
+        }
+
+        char *json_string = cJSON_Print(json);
+
+        CURL *curl;
+        CURLcode cURLCode;
+        curl = curl_easy_init();
+        struct MemoryStruct chunk;
+        chunk.memory = malloc(1);
+        chunk.size = 0;
+        if(curl) {
+            char *uri;
+            asprintf(&uri, "http://%s:%s/ntlm/auth", arguments.api_host, arguments.api_port);
+            curl_easy_setopt(curl, CURLOPT_WRITEFUNCTION, write_callback);
+            curl_easy_setopt(curl, CURLOPT_WRITEDATA, &chunk);
+
+            curl_easy_setopt(curl, CURLOPT_CUSTOMREQUEST, "POST");
+            curl_easy_setopt(curl, CURLOPT_URL, uri);
+            curl_easy_setopt(curl, CURLOPT_FOLLOWLOCATION, 1L);
+            curl_easy_setopt(curl, CURLOPT_DEFAULT_PROTOCOL, "https");
+            struct curl_slist *headers = NULL;
+            headers = curl_slist_append(headers, "Content-Type: application/json");
+            curl_easy_setopt(curl, CURLOPT_HTTPHEADER, headers);
+            curl_easy_setopt(curl, CURLOPT_POSTFIELDS, json_string);
+            cURLCode = curl_easy_perform(curl);
+            if (cURLCode == CURLE_OK) {
+                status = 0;
+                long http_response_code;
+                curl_easy_getinfo(curl, CURLINFO_HTTP_CODE, &http_response_code);
+                if (http_response_code != 200) {
+                    status = 1;  // consider non-200 response as auth failures.
+                }
+                printf("%s\n", chunk.memory);
+            } else {
+                status = SIGTERM; // timeout or any network errors, considered as time-outs
+                fprintf(stderr, "curl_easy_perform() failed: %s\n", curl_easy_strerror(cURLCode));
```

Comment:
why is status set to 1 there too?

Reply:
in old version, we consider SIGTERM as "timeout -killed" and all other codes as authentication failures. 

I changed it to : 
CURLE_OK && HTTP 200 -> status = 0 
CURLE_OP_TIMEOUT / RESOLV FAIL / CONNECT FAIL to "timeout", which will set the "status" to SIGTERM 
and keep the status = http resp code. 
need to test if it's supported in statusd.